src/components/TableNew/TableRow.tsx

```diff
@@ -0,0 +1,60 @@
+import { FC, TableHTMLAttributes, useCallback, useContext } from 'react'
+import { cn } from '@/lib/utils'
+import { TableContext } from '@/shared/context/TableContext/TableContext'
+import { TableActionsContext } from '@/shared/context/TableContext/TableActionsContext'
+import { Tooltip } from '@/components/Tooltip'
+import { Paragraph } from '@/components/TypographyNew'
+
+interface TableRowProps extends Omit<TableHTMLAttributes<HTMLTableRowElement>, 'onClick'> {
+  rowId: string
+  onClick?: (rowId: string) => void
+}
+
+export const TableRow: FC<TableRowProps> = ({ className, rowId, onClick, children, ...props }) => {
+  const tableContext = useContext(TableContext)
+  const dispatch = useContext(TableActionsContext)
+
+  const isSelected = tableContext ? !!tableContext.selectedRows[rowId] : false
+
+  const handleClick = useCallback(() => {
+    if (rowId) {
+      if (onClick) {
+        onClick(rowId)
+      }
+      if (dispatch) {
+        dispatch({ type: 'TOGGLE_ROW_SELECTION', payload: rowId })
+      }
```

Comment:
in theory, we should probably first dispatch then notify the parent (`onClick`) in case any processes depend on the updated state.